File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
    
        pairs = { ')': '(', ']': '[', '}': '{' }
        stack = []
        for i in s:
            if i in pairs.values():
                stack.append(i)
            elif i in pairs.keys():
                if len(stack) == 0:
                    return False
                elif stack[-1] == pairs[i]:
                    stack.pop()
                else:
                    return False
        
        return len(stack) == 0
            

File: Data Structures & Algorithms/minimum-stack/submission-2.cpp
class MinStack {
public:
    std::stack<int> st, minstack;
    MinStack() {
    }
    
    void push(int val) {
        st.push(val);
        if ((minstack.size() == 0 ) || (val <= minstack.top())) {
            minstack.push(val);
        }
        
    }
    
    void pop() {
        int x = st.top();
        st.pop();
        if (x == minstack.top()) {
            minstack.pop();
        }
        
    }
    
    int top() {
        return st.top();
        
    }
    
    int getMin() {
        return minstack.top();
        
    }
};
